libs/arangodb/langchain_arangodb/chains/graph_qa/arangodb.py

```diff
@@ -427,11 +427,22 @@ def _call(
         if max_history_messages <= 0:
             raise ValueError("max_history_messages must be greater than 0")
 
+        if self.chat_history_store is not None:
+            collection_name = self.chat_history_store._collection_name  # type: ignore
+
         chat_history = []
-        if include_history and self.chat_history_store is not None:
-            for msg in self.chat_history_store.messages[-max_history_messages:]:
-                cls = HumanMessage if msg.type == "human" else AIMessage
-                chat_history.append(cls(content=msg.content))
+        if include_history:
+            aql = f"""
+                FOR doc IN {collection_name}
+                SORT doc._key DESC
+                LIMIT @n
+                RETURN UNSET(doc, ["_id", "_key", "_rev", "session_id"])
+                """
```

Comment:
Small adjustment to this query, could we wrap this in `def get_messages_by_role` and store it in the ChatHistory class?

```suggestion
            aql = f"""
                FOR doc IN {collection_name}
                    FILTER doc.session_id == @session_id
                    AND doc.role == @role
                    SORT doc._key DESC
                    LIMIT @n
                    RETURN UNSET(doc, ["_id", "_key", "_rev", "session_id"])
                """
```

```py
if include_history:
    chat_history.extend(self.chat_history_store.get_messages_by_role("qa"))
```

Reply:
I just realized the reason I didn't wrap this in `get_messages_by_role()` here is because we want to include all messages i.e. both qa and human message(feedback). Then I thought of wrapping this with `messages()` but it wouldn't recognize qa message as it's not `BaseMessage` instance. 

Therefore, I add `get_messages()` that optionally takes a `role` argument. If `role` is given, it retrieves messages by role otherwise it returns all messages. What do you think?

Reply:
This makes sense agreed! Thanks